thoughts/plans/pi-glm-5-2-routing-cost-plan.html

```
@@ -0,0 +1,385 @@
+    <ol>
+      <li>Do not change `install.sh` Pi defaults away from GPT-5.5 medium for this pilot.</li>
+      <li>Update `install.sh` so `enabledModels` explicitly includes both `openai-codex/gpt-5.5` and `opencode/glm-5.2`, while preserving `defaultProvider: openai-codex` and `defaultModel: gpt-5.5`.</li>
+      <li>Update `scripts/verify-pi-install.sh` to assert the installed `~/.pi/agent/settings.json` has unchanged GPT-5.5 defaults and contains both `openai-codex/gpt-5.5` and `opencode/glm-5.2` in `enabledModels`.</li>
+      <li>Update `_pi/prompts/run-plan.md`, `_pi/prompts/dev:run.md`, `_pi/prompts/dev:plan.md`, or wrapper guidance to describe both invocation paths for GLM planning: operator switches the active scoped model to GLM-5.2 high before planning, or the GPT-5.5 default parent delegates a bounded planning/orchestration packet to <code>orchestrator-glm</code>.</li>
+      <li>Document the default-vs-scoped model decision and planning invocation paths in `_pi/README.md` and relevant prompt guidance.</li>
+    </ol>
+    <h3>Verify</h3>
+    <pre><code>rg -n "model: openai/gpt-5.5|DEFAULT_PROVIDER|DEFAULT_MODEL|enabledModels|opencode/glm-5.2|glm-5.2" _pi install.sh scripts/verify-pi-install.sh
+./install.sh --pi
+./scripts/verify-pi-install.sh
+python3 - <<'PY'
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ < ].

(spec-char-escape)